### PR TITLE
Viite 2358 fix to production

### PR DESCRIPTION
**Pull request by andre-policarpo**
**Command used**: git cherry-pick --ff VIITE-2358_split_link_to_have_same_roadway_numbers_amount_on_each_track_section ^master

**Conflicts resolved on**:
- VIITE-2360 added creation for link if new
- VIITE-2360 fixes to have update in connected link id.
- VIITE-2384 removed commented code

**Note**:
Added one last commit, to handle some changes since we're not including _VIITE-2344_ - but when this task goes up, this last commit must be reverted. Shouldn't be better to **merge** also _VIITE-2344_?

### CODE REVIEW

**Comment by skosonen** [comment]
> **Note**:
> Added one last commit, to handle some changes since we're not including _VIITE-2344_ - but when this task goes up, this last commit must be reverted. Shouldn't be better to **merge** also _VIITE-2344_?

No need to revert any commits since this branch will not be merged anywhere. We just do the code review and then create a tag for the hotfix. Later when we deploy new version from the master, this last commit will be overridden by that code from the master (where there is no this commit).

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/ProjectLinkDAO.scala [hunk 1]
```diff
@@ -376,39 +382,41 @@ class ProjectLinkDAO {
         addressPS.setLong(9, pl.originalStartAddrMValue)
         addressPS.setLong(10, pl.originalEndAddrMValue)
         addressPS.setString(11, pl.createdBy.orNull)
-        addressPS.setDouble(12, CalibrationCode.getFromAddress(pl).value)
-        addressPS.setLong(13, pl.status.value)
-        addressPS.setLong(14, pl.roadType.value)
+        addressPS.setString(12, pl.createdBy.orNull)
```
Comment (skosonen) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/ProjectLinkDAO.scala:385]: Why do we put creator as modifier here?

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/package.scala [hunk 2]
```diff
@@ -32,6 +32,8 @@ package object viite {
 
   val MinDistanceForGeometryUpdate = 0.5
 
+  val MaxThreshHoldDistance = 2
```
Comment (skosonen) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/package.scala:35]: Fix speling mistake:
MaxThreshHoldDistance -> MaxThresholdDistance

Comment (skosonen) [reply digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/package.scala:35]: (This kind of minor things can be fixed to the pull request that is for the master branch.)

**Comment by andre-policarpo** [comment]
> o

**Pull request closed without merging**